src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -82,31 +85,36 @@ def isEnabled(self, service):
         return enabled
 
     @private
-    def tryConnect(self, host, port):
+    def tryConnect(self, host, port, fnldap):
         max_tries = 3
         connected = False
 
-        timeout = _fs().middlewared.plugins.service_monitor.socket_timeout
+        if self.name == 'activedirectory':
+            host_list = []
 
-        for i in range(0, max_tries):
-            # XXX What about UDP?
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(timeout)
-
-            try:
-                s.connect((host, port))
-                connected = True
+            for i in range(0, max_tries):
+                # Make max_tries attempts to get SRV records from DNS
+                host_list = fnldap.get_ldap_servers(host)
+                if host_list:
```

Comment:
So, who is the `host` here? Originally that was one of the DCs, but `get_ldap_servers()` expects `domain` instead:
```    @staticmethod
    def get_ldap_servers(domain, site=None):
        dcs = []
        if not domain:
            return dcs

        host = "_ldap._tcp.%s" % domain
        if site:
            host = "_ldap._tcp.%s._sites.%s" % (site, domain)

        dcs = FreeNAS_ActiveDirectory_Base.get_SRV_records(host)
        return dcs
```

Reply:
Host is ultimately from freenas-v1.db here:
sqlite> SELECT sm_host FROM services_servicemonitor ;
`sm_host = THEGIBSON.LAN`
```
            thread = ServiceMonitorThread(
                id=s['id'], frequency=s['sm_frequency'], retry=s['sm_retry'],
                host=s['sm_host'], port=s['sm_port'], name=thread_name,
                logger=self.logger, middleware=self.middleware
            )
```